Made login state persistent

mcsimps2 opened the pull request:
The login state persistence is as follows:

-When a user sends the app to the background and is currently in a session, the user "leaves" the session and then rejoins it once the app comes back into view, if it ever does.

-When a user sends the app to the background, is currently in session, and then removes the process from multitasking, the user leaves the session, and when the app re-opens, the user is still logged in but is presented at the main view with the "join session" textfield.

-When the user sends the app to the background, is not in a session, is logged in, and removes the process from multitasking, the user will remain logged in and will return to the main view with the join session textfield.

-When the user sends the app to the background and is not logged in, then the user will always return to the login screen, even if the process is removed from multitasking.

I hope that made sense. Let me know if I can clarify anything.

Resolves: #3

reesjones commented:
What if a user is in a session, sends app to background but does NOT remove from memory, and returns to the app? It should keep the user in the session.

mcsimps2 commented:
Yeah, I think the ideal would be
-in background: stay part of session
-removed from memory: removed from session

We'll have to explore ways to achieve that.  To look at:
-Services
-Keep alive signals
-Detecting closed connections on server end (TCP Fin)
